streaming/base/dataset.py

```diff
@@ -287,6 +288,9 @@ def __init__(self,
             raise ValueError(
                 'You must provide either `streams` or `remote`/`local`, but not both.')
 
+        # Create a temporary directory for the default stream
+        if remote is None and local is None and split is None:
+            local = mkdtemp()
```

Comment:
and by `remote`/`local`, we mean our API here is that at least one of them must be set

`mkdtemp` relies on a seed not internal to streaming so is not really usable

Reply:
This is added because in the code path where `streams` is provided, a default stream can get created with remote/local/split all as None which raises ValueError(f'Could not create a local directory. Specify a local directory with the `local` value.') in stream.py. We need a valid temporary directory and that directory doesn't get used so it should be fine.

Reply:
lol that is gnarly

what do you think about passing `default` as a dict instead of instantiating a "default" stream?

Reply:
Makes sense